nntrainer/tensor/weight.h

```diff
@@ -79,11 +79,11 @@ class Weight : public Var_Grad {
    * @param train If the variable is trainable
    * @param name Name for this weight
    */
-  Weight(
+  explicit Weight(
     const TensorDim &dim,
     const WeightInitializer init = WeightInitializer::WEIGHT_XAVIER_UNIFORM,
     const WeightRegularizer reg = WeightRegularizer::NONE,
-    const float reg_const = 1.0f, bool train = true, bool alloc_now = true,
+    const float reg_const = 1.0f, bool train = true, bool alloc_now = false,
```

Comment:
. add brief @param about alloc_now. 
. I wonder the reason of false default value is because to track the weight in manager??
  Could you elaborate why manager does not allow manager to track initialized var_grads?
  and so.. we cannot allocate buffer memory during the initialization of layer??

Reply:
> . add brief @param about alloc_now.

Added.

> . I wonder the reason of false default value is because to track the weight in manager??
Could you elaborate why manager does not allow manager to track initialized var_grads?
and so.. we cannot allocate buffer memory during the initialization of layer??

The default is false to ensure the same behavior of inouts+derivatives and weights+gradients. Earlier, weights were eagerly allocated, but others were not. Now, all 4 (weights/grads/inouts/derivs) wrappers are allocated without memory first, and they are allocated whenever required.
Manager allows tracking initialized var_grads. Further, as alloc_now is still a parameter that allows creating weights/var_grads which allocate memory at that moment.
Layers are allowed to allocate memory for weight in their init by passing `allow_now = true`, but making its default value to false means that nntrainer encourages to use false and let the manager handle allocation appropriately than allocating upon init.